### ISSUE TITLE
@SerialInfo annotations missing from generated enum serializers

### ISSUE THREAD

**Issue by tKe** [Kotlin/kotlinx.serialization#2183]
**Describe the bug**
`@SerialInfo` annotations are not present in the descriptor of enums when generated in kotlin 1.7.20+ or 1.8.0 and consumed from kotlix-serialization 1.4.1 or 1.5.0-RC.
They are however present when generated using 1.7.10 (and consumed with 1.4.1 / 1.5.0-RC) or consumed with 1.4.0 (when produced with Kotlin 1.7.20+ and 1.8.0).

This also seems to be expected given the test cases currently in [`EnumDescriptorsTest`](https://github.com/Kotlin/kotlinx.serialization/blob/4a7ada862186368d72086cf01c2359e118ee232c/core/commonTest/src/kotlinx/serialization/EnumDescriptorsTest.kt#L101-L102)

**To Reproduce**
```kotlin
import kotlinx.serialization.ExperimentalSerializationApi
import kotlinx.serialization.SerialInfo
import kotlinx.serialization.Serializable
import kotlinx.serialization.descriptors.SerialDescriptor

@SerialInfo
@Target(AnnotationTarget.CLASS)
annotation class Tag(val value: String)

@Serializable
@Tag("test")
enum class TaggedEnum { Sugar, Fudge }

fun main() {
    fun SerialDescriptor.tag() = annotations.filterIsInstance<Tag>().firstOrNull()?.value

    println(TaggedEnum.serializer().descriptor.tag())
}

```

**Expected behavior**
Annotations annotated with `@SerialInfo` should be present in the `SerialDescriptor.annotations` collection on generated Enum serializers as in previous versions. This currently breaks the ability to annotated an enum with Avro4K's `@AvroNamespace`.

**Environment**
 - Kotlin version: 1.7.20+ and 1.8.0
 - Library version: 1.4.1 (and 1.5.0-RC)
 - Kotlin platforms: JVM
 - Gradle version: 7.6

**Comment by shanshin** [comment]
Hi,
duplicates issue #2121, fixed in Kotlin `1.8.20`

**Comment by tKe** [comment]
I wasn't sure if it was a duplicate or not, given that with `kotlinx-serialization` `1.4.0` and kotlin `1.8.0` this works fine (which is our current workaround).
